tests/pfcwd/test_pfcwd_function.py

```diff
@@ -0,0 +1,518 @@
+import datetime
+import logging
+import os
+import pytest
+import time
+
+from common.fixtures.conn_graph_facts import fanout_graph_facts
+from common.helpers.assertions import pytest_assert
+from common.helpers.pfc_storm import PFCStorm
+from common.plugins.loganalyzer.loganalyzer import LogAnalyzer
+from files.pfcwd_helper import start_wd_on_ports
+from ptf_runner import ptf_runner
+
+TEMPLATES_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "templates")
+EXPECT_PFC_WD_DETECT_RE = ".* detected PFC storm .*"
+EXPECT_PFC_WD_RESTORE_RE = ".*storm restored.*"
+WD_ACTION_MSG_PFX = { "dontcare": "Verify PFCWD detection when queue buffer is not empty and proper function of drop action",
+                      "drop": "Verify proper function of drop action",
+                      "forward": "Verify proper function of forward action"
+                    }
+
+pytestmark = [pytest.mark.disable_loganalyzer]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope='function', autouse=True)
+def stop_pfcwd(duthost):
+    """
+    Fixture that stops PFC Watchdog before each test run
+
+    Args:
+        duthost(AnsibleHost) : dut instance
+    """
+    logger.info("--- Stop Pfcwd --")
+    duthost.command("pfcwd stop")
```

Comment:
do we want to record the pfcwd state here and restore it once tests are run?

Reply:
i have a fixture that will restore the pfcwd to running state at the end of the test run because its enabled by default on our testbeds. but each test should start with a clean slate, hence i do the stop before each testcase